### PR TITLE
Merge Dev to Main

### PR DESCRIPTION
**Pull request by danielprypliasko**

### CODE REVIEW

**Comment by cosmo1809** [comment]
**Have the requirements been met?**
All the requirements of the assignment have been met. The properties and methods of the die class are all correctly implemented, the game class rolls 3, 6 sided dice and returns the total, and the testing class performs the tests given in the brief with appropriate responses.
**Is the code formatted correctly & easy to read?**
Both PascalCase and XML comments have been used throughout which ensures the code is correctly formatted and easily read.
**Are different errors handled correctly?**
The test class ensures that all the code runs successfully and all possible errors are handled efficiently.

**Review by cosmo1809 — APPROVED** [review]
Have the requirements been met?
All the requirements of the assignment have been met. The properties and methods of the die class are all correctly implemented, the game class rolls 3, 6 sided dice and returns the total, and the testing class performs the tests given in the brief with appropriate responses.
Is the code formatted correctly & easy to read?
Both PascalCase and XML comments have been used throughout which ensures the code is correctly formatted and easily read.
Are different errors handled correctly?
The test class ensures that all the code runs successfully and all possible errors are handled efficiently.

File: CMP1903_A1_2324/Die.cs [hunk 1]
```diff
@@ -13,10 +13,22 @@ internal class Die
          * and one method that rolls the die, returns and integer and takes no parameters.
          */
 
-        //Property
+        //Properties
+        public int Value {  get; private set; }
+        private static Random _random = new Random();
```
Comment (eth23git) [inline CMP1903_A1_2324/Die.cs:18]: Good adherence to the style guides.

File: CMP1903_A1_2324/Game.cs [hunk 2]
```diff
@@ -15,7 +15,66 @@ internal class Game
...
+           foreach (Die d in DieList)
+           {
+                d.Roll();
+           }
+        }
+
+        // <summary>
+        // Sums up the value of all the Die objects and saves it in the Total property
+        // </summary>
+        public void Sum()
```
Comment (eth23git) [inline CMP1903_A1_2324/Game.cs:51]: Extensive use of XML commenting.

File: CMP1903_A1_2324/Testing.cs [hunk 3]
```diff
@@ -15,6 +16,59 @@ internal class Testing
          * Use debug.assert() to make the comparisons and tests.
          */
 
-        //Method
+        //Properties
+        private Game _testGame = new Game();
```
Comment (eth23git) [inline CMP1903_A1_2324/Testing.cs:20]: Good demonstration of when to use encapsulation.

File: CMP1903_A1_2324/Program.cs [hunk 4]
```diff
@@ -14,6 +14,21 @@ static void Main(string[] args)
              * Create a Game object and call its methods.
              * Create a Testing object to verify the output and operation of the other classes.
              */
+
+            // Instantiate a Game object
+            Game game = new Game();
```
Comment (eth23git) [inline CMP1903_A1_2324/Program.cs:19]: Good demonstration of instantiation.

**Review by eth23git — COMMENTED** [review]
The code is very well commented with XML and single line commenting. Style guides were clearly adhered to at all times. The requirements are well met and error handling and extensive testing are well implemented.

**Pull request merged**